packages/firebase_ml_vision/lib/src/barcode_detector.dart

```diff
@@ -4,18 +4,452 @@
   Future<void> detectInImage(FirebaseVisionImage visionImage) async {
-    // TODO: implement detectInImage
+    final List<dynamic> reply = await FirebaseVision.channel.invokeMethod(
+      'BarcodeDetector#detectInImage',
+      visionImage.imageFile.path,
+    );
+
+    final List<VisionBarcode> barcodes = <VisionBarcode>[];
+    reply.forEach((dynamic barcode) {
+      barcodes.add(new VisionBarcode._(barcode));
+    });
+
+    return barcodes;
   }
 }
 
-class BarcodeDetectorOptions {}
+class VisionBarcode {
+  final Rectangle<int> boundingBox;
+  final String rawValue;
+  final String displayValue;
+  final VisionBarcodeFormat format;
+  final List<Point<int>> cornerPoints;
+  final VisionBarcodeValueType valueType;
+  final VisionBarcodeEmail email;
+  final VisionBarcodePhone phone;
+  final VisionBarcodeSMS sms;
+  final VisionBarcodeURLBookmark url;
+  final VisionBarcodeWiFi wifi;
+  final VisionBarcodeGeoPoint geoPoint;
+  final VisionBarcodeContactInfo contactInfo;
+  final VisionBarcodeCalendarEvent calendarEvent;
+  final VisionBarcodeDriverLicense driverLicense;
+
+  VisionBarcode._(Map<dynamic, dynamic> _data)
+      : boundingBox = Rectangle<int>(
+          _data['left'],
+          _data['top'],
+          _data['width'],
+          _data['height'],
+        ),
+        rawValue = _data['raw_value'] != null ? _data['raw_value'] : null,
+        displayValue =
+            _data['display_value'] != null ? _data['display_value'] : null,
+        format = VisionBarcodeFormat._(_data['format']),
+        cornerPoints = _data['points'] == null
+            ? null
+            : _data['points']
+                .map<Point<int>>((dynamic item) => Point<int>(
+                      item[0],
+                      item[1],
+                    ))
+                .toList(),
+        valueType =
+            VisionBarcodeValueType.values.elementAt(_data['value_type']),
+        email = _data['email'] == null
+            ? null
+            : VisionBarcodeEmail._(_data['email']),
+        phone = _data['phone'] == null
+            ? null
+            : VisionBarcodePhone._(_data['phone']),
+        sms = _data['sms'] == null ? null : VisionBarcodeSMS._(_data['sms']),
+        url = _data['url'] == null
+            ? null
+            : VisionBarcodeURLBookmark._(_data['url']),
+        wifi =
+            _data['wifi'] == null ? null : VisionBarcodeWiFi._(_data['wifi']),
+        geoPoint = _data['geo_point'] == null
+            ? null
+            : VisionBarcodeGeoPoint._(_data['geo_point']),
+        contactInfo = _data['contact_info'] == null
+            ? null
+            : VisionBarcodeContactInfo._(_data['contact_info']),
+        calendarEvent = _data['calendar_event'] == null
+            ? null
+            : VisionBarcodeCalendarEvent._(_data['calendar_event']),
+        driverLicense = _data['driver_license'] == null
+            ? null
+            : VisionBarcodeDriverLicense._(_data['driver_license']);
+}
+
+// ios:
```

Comment:
Instead of the links, you can just copy and paste their documentation, such as

'This option specifies the barcode formats that the library should detect.'

Reply:
Also Dartdoc documentation uses `///` instead of `//`